### ISSUE TITLE
Jest 29 async await transform

### ISSUE THREAD

**Issue by JiaLiPassion** [thymikee/jest-preset-angular#2010]
### Please do not submit this issue.

- [X] I understand

**Comment by ahnpnl** [comment]
Hi @JiaLiPassion, in v13, we removed the hardcoded tsconfig to be below `es2017` which triggers the issue. We have updated documentation https://thymikee.github.io/jest-preset-angular/docs/getting-started/installation#configuration as well as mentioned in `README` and `CHANGELOG` about the change.

As long as users use target `es2015` or `es2016` there won’t be any problems.

**Comment by JiaLiPassion** [comment]
@ahnpnl , thank you, and I saw the new README to mention this change, in `@angular/cli`, `es2017+` targets are supported, but `async/await` is still being downleveled to `Promise`, https://github.com/angular/angular-cli/blob/5a171ddff66ff366089616736baf7545fe44f570/packages/angular_devkit/build_angular/src/babel/presets/application.ts#L235

Should `jest-preset-angular` also apply this behavior?

**Comment by ahnpnl** [comment]
Ah ha thank you for the information. I think those babel plugins should be included in this preset then.

**Comment by JiaLiPassion** [comment]
@ahnpnl , Got it, thank you.

**Comment by JiaLiPassion** [comment]
@ahnpnl , I am trying the guide here https://thymikee.github.io/jest-preset-angular/docs/guides/using-with-babel
1. create a `babel.config.js`.

```
module.exports = function (api) {
  api.cache(true);

  const presets = ["@babel/preset-env"];
  const plugins = [
    "@babel/plugin-transform-async-to-generator",
    "@babel/plugin-proposal-async-generator-functions",
  ];

  return {
    presets,
    plugins,
  };
};

```

2.  update `jest.config.ts` like this.

```
import type {Config} from '@jest/types'

export default <Config.InitialOptions> {
  preset: 'jest-preset-angular',
      setupFilesAfterEnv: ['<rootDir>/setup-jest.ts'], transform: {
        '^.+\\.(ts|html)$': 'jest-preset-angular',
        '^.+\\.js$': 'babel-jest',
      },
}

```

And it reports 

```
    Jest failed to parse a file. This happens e.g. when your code or its dependencies use non-standard JavaScript syntax, or when Jest is not configured to support such syntax.
```

When try to parse `app.component.html`. 
Do I miss some steps?

**Comment by ahnpnl** [comment]
Hi I think the doc is outdated, you can check the proper configuration here https://github.com/thymikee/jest-preset-angular/tree/main/e2e/babel-support

**Comment by JiaLiPassion** [comment]
Yeah, I also tried that one, the same error occurs. it seems the e2e is not testing an angular app.

**Comment by ahnpnl** [comment]
I quickly made an example here #2036

**Comment by JiaLiPassion** [comment]
Thanks, this one works!
